lib/cluster.js

```diff
@@ -260,6 +260,9 @@ function masterInit() {
       return process.nextTick(setupSettingsNT, settings);
     initialized = true;
     schedulingPolicy = cluster.schedulingPolicy;  // Freeze policy.
+    if (settings.scheduler) {
+      settings.scheduler(handles);
```

Comment:
I agree with Colin that this is not a good approach: it leaks an implementation detail (the `handles` dictionary) into the public API.

Say we wanted to change it to a Map - we wouldn't be able to because that would break existing code.